File: A1/A1-021/A1-021.py
# --------------------------------------------------
# File Name : A1-021.py
# Problem   : Leap Year
# Author    : Worralop Srichainont
# Date      : 2026-02-08
# --------------------------------------------------

# Input
year = int(input())

# Output
if year < 1582 or year % 400 == 0 or (year % 4 == 0 and year % 100 != 0):
    print("yes")
else:
    print("no")

File: A1/A1-021/A1-021.cpp
#include<iostream>
using namespace std;

int main()
{
    // This makes std::cin and std::cout faster
    ios_base::sync_with_stdio(false); cin.tie(0);
    // Input
    int year;
    cin >> year;
    // Output
    if(year < 1582) { cout << "yes"; }
    else if(year % 400 == 0) { cout << "yes"; }
    else if(year % 4 == 0 && year % 100 != 0) { cout << "yes"; }
    else { cout << "no"; }
    return 0;
}